### PR TITLE
[CARBONDATA-1316] drop partition

### PR DESCRIPTION
**Pull request by lionelcao**
Drop partition function

### CODE REVIEW

**Comment by QiangCai** [comment]
please describe your modification

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 1]
```diff
@@ -320,7 +322,143 @@ case class AlterTableSplitPartitionCommand(splitPartitionModel: AlterTableSplitP
   }
 }
 
-case class CreateTable(cm: TableModel, createDSTable: Boolean = true) extends RunnableCommand
+case class AlterTableDropPartition(alterTableDropPartitionModel: AlterTableDropPartitionModel)
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: please comment different points between different partition type in drop partition command.

**Comment by lionelcao** [comment]
Refer to below guidance for illustration of drop partition function:
https://github.com/apache/carbondata/blob/master/docs/partition-guide.md#drop-a-partition

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/AlterTableLoadPartitionRDD.scala [hunk 2]
```diff
@@ -104,20 +103,20 @@ class AlterTableSplitPartitionRDD[K, V](
...
             } else {
+                val segmentProperties = PartitionUtils.getSegmentProperties(identifier,
+                    segmentId, partitionIds.toList, oldPartitionIdList, partitionInfo)
+                val processor = new RowResultProcessor(
+                    carbonTable,
+                    carbonLoadModel,
+                    segmentProperties,
+                    tempStoreLoc,
+                    bucketId
+                )
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/AlterTableLoadPartitionRDD.scala]: move to last line

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/DataManagementFunc.scala [hunk 3]
```diff
@@ -305,6 +305,27 @@ object DataManagementFunc {
     futureList.add(future)
   }
 
+  def executeDroppingPartition(sqlContext: SQLContext,
+      carbonLoadModel: CarbonLoadModel,
+      executor: ExecutorService,
+      storePath: String,
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/DataManagementFunc.scala]: we can get storePath from carbonLoadModel

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/PartitionDropper.scala [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import org.apache.carbondata.common.logging.LogServiceFactory
+import org.apache.carbondata.core.metadata.schema.partition.PartitionType
+import org.apache.carbondata.spark.{AlterPartitionResultImpl, PartitionFactory}
+
+object PartitionDropper {
+
+  val logger = LogServiceFactory.getLogService(PartitionDropper.getClass.getName)
+
+  def triggerPartitionDrop(dropPartitionCallableModel: DropPartitionCallableModel): Unit = {
+    val sc = dropPartitionCallableModel.sqlContext.sparkContext
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/PartitionDropper.scala]: better to remove sc from DropPartitionCallableModel.
suggest to package the parameters of AlterTableLoadPartitionRDD and CarbonScanPartitionRDD

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 5]
```diff
@@ -320,7 +322,143 @@ case class AlterTableSplitPartitionCommand(splitPartitionModel: AlterTableSplitP
...
+  extends RunnableCommand with DataProcessCommand with SchemaProcessCommand {
+  val LOGGER = LogServiceFactory.getLogService(this.getClass.getName)
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: why to add so many attribute?

File: processing/src/main/java/org/apache/carbondata/processing/spliter/CarbonDataSpliterUtil.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+package org.apache.carbondata.processing.spliter;
+
+import java.util.List;
+
+import org.apache.carbondata.common.CarbonIterator;
+import org.apache.carbondata.common.logging.LogService;
+import org.apache.carbondata.common.logging.LogServiceFactory;
+import org.apache.carbondata.core.scan.result.BatchResult;
+
+public final class CarbonDataSpliterUtil {
```
Comment (QiangCai) [inline processing/src/main/java/org/apache/carbondata/processing/spliter/CarbonDataSpliterUtil.java]: please remove it if no need.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/13/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala (hunk 5 above)
Comment (lionelcao) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: The variables could only get from sparkSession and both processSchema and processData will use those variables.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/121/

**Comment by ravipesala** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder1/19/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/755/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/139/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/13/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/201/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/78/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/833/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/79/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/202/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/80/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/203/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/834/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/835/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/83/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/206/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/838/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**